x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/cohere/rerank/CohereRerankTaskSettingsTests.java

```diff
@@ -149,6 +149,14 @@ protected CohereRerankTaskSettings createTestInstance() {
 
     @Override
     protected CohereRerankTaskSettings mutateInstance(CohereRerankTaskSettings instance) throws IOException {
-        return randomValueOtherThan(instance, CohereRerankTaskSettingsTests::createRandom);
+        var topNDocsOnly = instance.getTopNDocumentsOnly();
+        var returnDocuments = instance.getReturnDocuments();
+        var maxChunksPerDoc = instance.getMaxChunksPerDoc();
+        switch (randomInt(2)) {
+            case 0 -> topNDocsOnly = randomValueOtherThan(topNDocsOnly, () -> randomFrom(randomIntBetween(1, 10), null));
+            case 1 -> returnDocuments = randomValueOtherThan(returnDocuments, () -> randomFrom(randomBoolean(), null));
```

Comment:
If we end up going for a non-null value and the original value is also non-null, should we just negate original to avoid slowing down the test if we're getting very unlucky?